cmd/crates/soroban-test/tests/it/rpc_provider.rs

```diff
@@ -0,0 +1,97 @@
+use httpmock::{prelude::*, Mock};
+use serde_json::json;
+use soroban_rpc::{GetEventsResponse, GetNetworkResponse};
+use soroban_test::{TestEnv, LOCAL_NETWORK_PASSPHRASE};
+
+#[tokio::test]
+async fn test_use_rpc_provider_with_auth_header() {
+    // mock out http request to rpc provider
+    let server = MockServer::start();
+    let generate_account_mock = mock_generate_account(&server);
+    let get_network_mock = mock_get_network(&server);
+    let get_events_mock = mock_get_events(&server);
+
+    // create a new test environment with the mock server
+    let rpc_url = server.url("");
+    let rpc_headers = vec![("Authorization".to_string(), "Bearer test-token".to_string())];
+    let sandbox = &TestEnv::with_rpc_provider(&rpc_url, rpc_headers);
+
+    sandbox
```

Comment:
Should we check that the response contains expected JSON?

Reply:
It's a mock so not sure that would be of benefit?

Reply:
Good question - since this test is mocking out the request to the rpc provider, I think it's okay not to check that response since it will be the `GetEventsResponse` that we're telling the provider to return on line 91. But I'm happy to rethink that!